store/pools/index.ts

```diff
@@ -0,0 +1,65 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { BasePool } from '@/hooks/types';
+import type { PayloadAction } from '@reduxjs/toolkit';
+
+export interface PoolsState {
+  pools: BasePool[] | [];
+  trendingPools: BasePool[] | [];
+  hotPools: BasePool[] | [];
+  topGainers: BasePool[] | [];
```

Comment:
Why do you need the empty array in this type union?

```suggestion
  pools: BasePool[];
  trendingPools: BasePool[];
  hotPools: BasePool[];
  topGainers: BasePool[];
```

Reply:
That's a mistake @renanmav thanks for pointing it out. The PR is still a draft though. We're considering leaving the BE to handle these expensive pool and price fetches

Reply:
removed